pkg/database/database.go

```diff
@@ -130,6 +137,10 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
 		return fmt.Errorf("database gorm.Open: %w", err)
 	}
 
+	if c.MaxConnectionIdleTime > 0 {
```

Comment:
I haven't done as much research here, but wouldn't `SetConnMaxLifetime` work better (and avoid all this overhead)?

```go
db.DB().SetConnMaxLifetime(1*time.Minute)
```

Reply:
made that the default - we can start there and see if it helps